include/Beman/Optional26/optional.hpp

```diff
@@ -202,22 +203,39 @@ template <class T>
 class optional;
 
 } // namespace beman::optional
+
 namespace std {
+// Since P3168R1: Give std::optional Range Support.
 template <typename T>
 inline constexpr bool ranges::enable_view<beman::optional::optional<T>> = true;
 
+// Since ${PAPER_NUMBER}: ${PAPER_TITLE}.
+// Note: P3168 and P2988 may have different flows inside LEWG/LWG.
+// Implementation of the range support for optional<T&> reflects P3168R1 for now.
+// [optional.iterators], iterator support
 template <typename T>
 inline constexpr bool ranges::enable_view<beman::optional::optional<T&>> = true;
 
+// TODO: document why this is needed.
 template <typename T>
 inline constexpr bool ranges::enable_borrowed_range<beman::optional::optional<T*>> = true;
 
+// TODO: document why this is needed.
 template <typename T>
 inline constexpr bool ranges::enable_borrowed_range<beman::optional::optional<std::reference_wrapper<T>>> = true;
 
+// TODO: document why this is needed.
 template <typename T>
 inline constexpr bool ranges::enable_borrowed_range<beman::optional::optional<T&>> = true;
 
+// Since P3168R1: Give std::optional Range Support.
+// Note: At 13.06.2024, gcc and msvc do not implement P2585R1: "Improve default container formatting".
+// TODO @neatudarius: always enable once supported by other compilers.
+#if defined(__clang__)
```

Comment:
The right check here is for `__cpp_lib_format_ranges`

Reply:
thanks!